### PR TITLE
Fix diagnostics on MacOS and duplicated bases.

### PR DESCRIPTION
**Pull request by scturtle**

### CODE REVIEW

File: src/indexer.cc [hunk 1]
```diff
@@ -1222,6 +1222,7 @@ std::vector<std::unique_ptr<IndexFile>> Index(
       Uniquify(it.second.uses);
       // e.g. declaration + out-of-line definition
       Uniquify(it.second.def.funcs);
+      Uniquify(it.second.def.bases);
```
Comment (MaskRay) [inline src/indexer.cc]: swap this line with `funcs`?

**Comment by scturtle** [comment]
Yes for MacOS but no for linux. Even specified in `LoadFromCompilerInvocationAction` of `indexer.cc`.

**Comment by MaskRay** [comment]
Do you mean:

* For Mac OS X, `g_config->clang.resourceDir` needs to be passed to `ASTUnit::createFromCommandLine`
* For Linux, `-resource-dir` may be explicitly passed

?

**Comment by scturtle** [comment]
Yes.

**Comment by MaskRay** [comment]
Thanks!

**Pull request merged**